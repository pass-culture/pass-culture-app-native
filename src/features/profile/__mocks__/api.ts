export enum CHANGE_EMAIL_ERROR_CODE {
  TOKEN_EXISTS = 'TOKEN_EXISTS',
  INVALID_PASSWORD = 'INVALID_PASSWORD',
  EMAIL_UPDATE_ATTEMPTS_LIMIT = 'EMAIL_UPDATE_ATTEMPTS_LIMIT',
}
