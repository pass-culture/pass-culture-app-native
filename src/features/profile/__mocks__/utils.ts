export const isUserExBeneficiary = jest.fn().mockReturnValue(false)
export const isUserBeneficiary = jest.fn().mockReturnValue(true)
export const useIsUserUnderageBeneficiary = jest.fn().mockReturnValue(false)
export const isUserUnderageBeneficiary = jest.fn().mockReturnValue(false)
export const useIsUserUnderage = jest.fn().mockReturnValue(false)
export const matchSubscriptionMessageIconToSvg = jest
  .fn()
  .mockReturnValue('matchSubscriptionMessageIconToSvg')

export const handleCallToActionLink = jest.fn()

export const computeCredit = jest.fn().mockReturnValue(3)
