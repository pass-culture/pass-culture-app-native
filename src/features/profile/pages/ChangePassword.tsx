import { t } from '@lingui/macro'
import React, { useState } from 'react'
import styled from 'styled-components/native'

import {
  isPasswordCorrect,
  PasswordSecurityRules,
} from 'features/auth/components/PasswordSecurityRules'
import { useChangePasswordMutation } from 'features/auth/mutations'
import { _ } from 'libs/i18n'
import { ButtonPrimary } from 'ui/components/buttons/ButtonPrimary'
import { InputError } from 'ui/components/inputs/InputError'
import { PasswordInput } from 'ui/components/inputs/PasswordInput'
import { getSpacing, Spacer, Typo } from 'ui/theme'

import { ProfileHeaderWithNavigation } from '../components/ProfileHeaderWithNavigation'
import { ProfileContainer } from '../components/reusables'

export function ChangePassword() {
  const [currentPassword, setCurrentPassword] = useState('')
  const [newPassword, setNewPassword] = useState('')
  const [confirmedPassword, setConfirmedPassword] = useState('')

  const displayNotMatchingError = confirmedPassword.length > 0 && confirmedPassword !== newPassword

  const shouldSave =
    isPasswordCorrect(currentPassword) &&
    isPasswordCorrect(newPassword) &&
    confirmedPassword === newPassword

  const { mutate: changePassword, isLoading } = useChangePasswordMutation(() => {})

  function submitPassword() {
    changePassword({ currentPassword, newPassword })
  }

  return (
    <React.Fragment>
      <ProfileHeaderWithNavigation title={_(t`Mot de passe`)} />
      <Container>
        <StyledInput>
          <Typo.Body>{_(t`Mot de passe actuel`)}</Typo.Body>
          <Spacer.Column numberOfSpaces={2} />
          <PasswordInput
            value={currentPassword}
            autoFocus={true}
            onChangeText={setCurrentPassword}
            placeholder={_(/*i18n: password placeholder */ t`Ton mot de passe actuel`)}
          />
        </StyledInput>
        <Spacer.Column numberOfSpaces={5} />
        <StyledInput>
          <Typo.Body>{_(t`Nouveau mot de passe`)}</Typo.Body>
          <Spacer.Column numberOfSpaces={2} />
          <PasswordInput
            value={newPassword}
            onChangeText={setNewPassword}
            placeholder={_(/*i18n: password placeholder */ t`Ton mot de passe`)}
          />
          <PasswordSecurityRules password={newPassword} />
        </StyledInput>
        <Spacer.Column numberOfSpaces={5} />
        <StyledInput>
          <Typo.Body>{_(t`Confirmer le mot de passe`)}</Typo.Body>
          <Spacer.Column numberOfSpaces={2} />
          <PasswordInput
            value={confirmedPassword}
            onChangeText={setConfirmedPassword}
            placeholder={_(/*i18n: password placeholder */ t`Confirmer le mot de passe`)}
          />
        </StyledInput>
        <Spacer.Column numberOfSpaces={2} />
        <InputError
          visible={displayNotMatchingError}
          messageId="les mots de passe ne concordent pas"
          numberOfSpacesTop={0}
        />
        <Spacer.Flex flex={1} />
        <ButtonPrimary
          title={_(t`Enregistrer`)}
          onPress={submitPassword}
          disabled={!shouldSave || isLoading}
        />
      </Container>
    </React.Fragment>
  )
}

const StyledInput = styled.View({
  display: 'flex',
  flexDirection: 'column',
  alignItems: 'flex-start',
  width: '100%',
  maxWidth: getSpacing(125),
})

const Container = styled(ProfileContainer)({
  padding: getSpacing(4),
})
