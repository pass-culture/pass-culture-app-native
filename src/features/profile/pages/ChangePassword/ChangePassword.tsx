import { t } from '@lingui/macro'
import React, { useEffect, useRef, useState } from 'react'
import { Platform, ScrollView, StyleProp, ViewStyle } from 'react-native'
import { useSafeAreaInsets } from 'react-native-safe-area-context'
import styled from 'styled-components/native'
import { useTheme } from 'styled-components/native'
import { v4 as uuidv4 } from 'uuid'

import {
  isPasswordCorrect,
  PasswordSecurityRules,
} from 'features/auth/components/PasswordSecurityRules'
import { useChangePasswordMutation } from 'features/auth/mutations'
import { analytics } from 'libs/analytics'
import { theme } from 'theme'
import { ButtonPrimary } from 'ui/components/buttons/ButtonPrimary'
import { PageHeader } from 'ui/components/headers/PageHeader'
import { InputError } from 'ui/components/inputs/InputError'
import { PasswordInput } from 'ui/components/inputs/PasswordInput'
import { useForHeightKeyboardEvents } from 'ui/components/keyboard/useKeyboardEvents'
import { SNACK_BAR_TIME_OUT, useSnackBarContext } from 'ui/components/snackBar/SnackBarContext'
import { useEnterKeyAction } from 'ui/hooks/useEnterKeyAction'
import { getSpacing, Spacer } from 'ui/theme'
import { Form } from 'ui/web/form/Form'

export function ChangePassword() {
  const { isMobileViewport, isTouch } = useTheme()
  const { showSuccessSnackBar } = useSnackBarContext()

  const [currentPassword, setCurrentPassword] = useState('')
  const [newPassword, setNewPassword] = useState('')
  const [confirmedPassword, setConfirmedPassword] = useState('')
  const [hasError, setHasError] = useState(false)
  const [keyboardHeight, setKeyboardHeight] = useState(0)
  const [shouldDisplayPasswordRules, setShouldDisplayPasswordRules] = useState(false)
  const passwordDescribedBy = uuidv4()

  const displayNotMatchingError = confirmedPassword.length > 0 && confirmedPassword !== newPassword

  const shouldSave =
    isPasswordCorrect(currentPassword) &&
    isPasswordCorrect(newPassword) &&
    confirmedPassword === newPassword

  useForHeightKeyboardEvents(setKeyboardHeight)

  // on type, reset error field
  useEffect(() => {
    if (hasError) {
      setHasError(false)
    }
  }, [currentPassword, newPassword, confirmedPassword])

  const { mutate: changePassword, isLoading } = useChangePasswordMutation(
    () => {
      setCurrentPassword('')
      setNewPassword('')
      setConfirmedPassword('')
      setShouldDisplayPasswordRules(false)
      showSuccessSnackBar({
        message: t`Mot de passe modifiÃ©`,
        timeout: SNACK_BAR_TIME_OUT,
      })
      analytics.logHasChangedPassword('changePassword')
    },
    () => {
      setHasError(true)
      setShouldDisplayPasswordRules(false)
    }
  )

  function submitPassword() {
    changePassword({ currentPassword, newPassword })
  }

  function updateNewPassword(value: string) {
    if (!shouldDisplayPasswordRules) {
      setShouldDisplayPasswordRules(true)
    }
    setNewPassword(value)
  }

  const scrollRef = useRef<ScrollView | null>(null)

  const { bottom } = useSafeAreaInsets()

  const disabled = !shouldSave || isLoading
  const passwordInputErrorId = uuidv4()
  const passwordConfirmationErrorId = uuidv4()

  useEnterKeyAction(!disabled ? submitPassword : undefined)

  return (
    <Container>
      <PageHeader title={t`Mot de passe`} />
      <Spacer.TopScreen />
      <StyledScrollView
        ref={scrollRef}
        contentContainerStyle={getScrollViewContentContainerStyle(keyboardHeight)}
        keyboardShouldPersistTaps="handled">
        <Spacer.Column numberOfSpaces={20} />
        <Form.MaxWidth>
          <PasswordInput
            label={t`Mot de passe actuel`}
            value={currentPassword}
            autoFocus={true}
            onChangeText={setCurrentPassword}
            placeholder={t`Ton mot de passe actuel`}
            isRequiredField
            accessibilityDescribedBy={passwordInputErrorId}
          />
          <InputError
            visible={hasError}
            messageId={t`Mot de passe incorrect`}
            numberOfSpacesTop={0}
            relatedInputId={passwordInputErrorId}
          />
          <Spacer.Column numberOfSpaces={7} />
          <PasswordInput
            label={t`Nouveau mot de passe`}
            accessibilityDescribedBy={passwordDescribedBy}
            value={newPassword}
            onChangeText={updateNewPassword}
            placeholder={t`Ton nouveau mot de passe`}
            isRequiredField
          />
          {!!(shouldDisplayPasswordRules && newPassword.length > 0) && (
            <PasswordSecurityRules password={newPassword} nativeID={passwordDescribedBy} />
          )}
          <Spacer.Column numberOfSpaces={5} />
          <PasswordInput
            label={t`Confirmer le mot de passe`}
            value={confirmedPassword}
            onChangeText={setConfirmedPassword}
            placeholder={t`Confirmer le mot de passe`}
            onFocus={() => {
              setTimeout(() => scrollRef?.current?.scrollToEnd({ animated: true }), 60)
            }}
            isRequiredField
            accessibilityDescribedBy={passwordConfirmationErrorId}
          />
          <InputError
            visible={displayNotMatchingError}
            messageId={t`Les mots de passe ne concordent pas`}
            numberOfSpacesTop={2}
            relatedInputId={passwordConfirmationErrorId}
          />

          {isMobileViewport && isTouch ? (
            <Spacer.Flex flex={1} />
          ) : (
            <Spacer.Column numberOfSpaces={10} />
          )}

          {!!keyboardHeight && <Spacer.Column numberOfSpaces={2} />}
          <ButtonContainer paddingBottom={keyboardHeight ? 0 : bottom}>
            <ButtonPrimary
              wording={t`Enregistrer`}
              accessibilityLabel={t`Enregistrer les modifications`}
              onPress={submitPassword}
              disabled={disabled}
            />
          </ButtonContainer>
        </Form.MaxWidth>
        <Spacer.Column numberOfSpaces={6} />
      </StyledScrollView>
    </Container>
  )
}

const Container = styled.View(({ theme }) => ({
  flex: 1,
  backgroundColor: theme.colors.white,
}))

const getScrollViewContentContainerStyle = (keyboardHeight: number): StyleProp<ViewStyle> => ({
  flexGrow: 1,
  flexDirection: 'column',
  paddingBottom: Platform.OS === 'ios' ? keyboardHeight : 0,
  backgroundColor: theme.colors.white,
  alignItems: 'center',
})

const ButtonContainer = styled.View<{ paddingBottom: number }>(({ paddingBottom }) => ({
  paddingBottom,
  alignItems: 'center',
  width: '100%',
}))

const StyledScrollView = styled(ScrollView)({
  paddingHorizontal: getSpacing(5.5),
})
