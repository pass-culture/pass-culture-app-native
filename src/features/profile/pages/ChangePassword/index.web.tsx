import loadable from '@loadable/component'
import React from 'react'

import { LoadingPage } from 'ui/components/LoadingPage'

export const ChangePassword = loadable(() => import('./ChangePassword'), {
  resolveComponent: (components) => components.ChangePassword,
  fallback: <LoadingPage />,
})
