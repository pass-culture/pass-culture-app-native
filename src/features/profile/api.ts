import { useMutation, useQueryClient } from 'react-query'

import { api } from 'api/api'
import { UserProfileResponse, UserProfileUpdateRequest } from 'api/gen'
import { QueryKeys } from 'libs/queryKeys'

export enum CHANGE_EMAIL_ERROR_CODE {
  TOKEN_EXISTS = 'TOKEN_EXISTS',
  INVALID_PASSWORD = 'INVALID_PASSWORD',
  EMAIL_UPDATE_ATTEMPTS_LIMIT = 'EMAIL_UPDATE_ATTEMPTS_LIMIT',
}

export interface ChangeEmailRequest {
  email: string
  password: string
}

type ResetRecreditAmountToShowMutationOptions = {
  onSuccess: () => void
  onError: (error: unknown) => void
}

export function useUpdateProfileMutation(
  onSuccessCallback: (data: UserProfileResponse) => void,
  onErrorCallback: (error: unknown) => void
) {
  const client = useQueryClient()
  return useMutation((body: UserProfileUpdateRequest) => api.postnativev1profile(body), {
    onSuccess(response: UserProfileResponse) {
      client.setQueryData(QueryKeys.USER_PROFILE, (old: UserProfileResponse | undefined) => ({
        ...(old || {}),
        ...response,
      }))
      onSuccessCallback(response)
    },
    onError: onErrorCallback,
  })
}

export function useResetRecreditAmountToShow({
  onSuccess,
  onError,
}: ResetRecreditAmountToShowMutationOptions) {
  return useMutation(() => api.postnativev1resetRecreditAmountToShow(), {
    onSuccess,
    onError,
  })
}
