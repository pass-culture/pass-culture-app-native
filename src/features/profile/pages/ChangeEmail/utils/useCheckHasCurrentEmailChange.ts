import { useQuery } from 'react-query'

import { api } from 'api/api'
import { QueryKeys } from 'libs/queryKeys'

export const useCheckHasCurrentEmailChange = () => {
  const { data: currentEmailChangeTimestampResponse } = useQuery(
    QueryKeys.EMAIL_CHANGE_EXPIRATION_TIMESTAMP,
    () => api.getnativev1profiletokenExpiration()
  )
  const hasCurrentEmailChange = !!currentEmailChangeTimestampResponse?.expiration
  return { hasCurrentEmailChange }
}
