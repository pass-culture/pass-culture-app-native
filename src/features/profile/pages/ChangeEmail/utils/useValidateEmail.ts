import { t } from '@lingui/macro'

import { useUserProfileInfo } from 'features/home/api'
import { isEmailValid } from 'ui/components/inputs/emailCheck'

export const useIsCurrentUserEmail = (email: string): boolean => {
  const { data: user } = useUserProfileInfo()
  return email.toLowerCase() === user?.email?.toLowerCase()
}

export function useValidateEmail(email: string): useValidateEmailReturn {
  const isCurrentUserEmail = useIsCurrentUserEmail(email)

  if (email.length === 0) return { emailErrorMessage: null, isEmailValid: false }
  if (!isEmailValid(email))
    return {
      emailErrorMessage: t`L'e-mail renseigné est incorrect. Exemple de format attendu\u00a0: edith.piaf@email.fr`,
      isEmailValid: false,
    }
  if (isCurrentUserEmail)
    return {
      emailErrorMessage: t`L'e-mail saisi est identique à votre e-mail actuel`,
      isEmailValid: false,
    }
  return { emailErrorMessage: null, isEmailValid: true }
}

type useValidateEmailReturn = { emailErrorMessage: string | null; isEmailValid: boolean }
