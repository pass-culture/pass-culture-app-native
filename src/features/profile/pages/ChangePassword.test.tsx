import { render, act, fireEvent, waitFor } from '@testing-library/react-native'
import { rest } from 'msw'
import React from 'react'
import waitForExpect from 'wait-for-expect'

import { ChangePasswordRequest } from 'api/gen'
import { env } from 'libs/environment'
import { EmptyResponse } from 'libs/fetch'
import { reactQueryProviderHOC } from 'tests/reactQueryProviderHOC'
import { server } from 'tests/server'
import { flushAllPromises } from 'tests/utils'
import { displaySuccessSnackBar } from 'ui/components/snackBar/__mocks__/SnackBarContext.ts'
import { SNACK_BAR_TIME_OUT, useSnackBarContext } from 'ui/components/snackBar/SnackBarContext'
import { ColorsEnum } from 'ui/theme'

import { ChangePassword } from './ChangePassword'

const mockedUseSnackBarContext = useSnackBarContext as jest.Mock

jest.mock('ui/components/snackBar/SnackBarContext', () => ({
  useSnackBarContext: jest.fn(() => ({})),
}))

async function renderChangePassword() {
  const wrapper = render(reactQueryProviderHOC(<ChangePassword />))
  await act(async () => {
    await flushAllPromises()
  })
  return wrapper
}

describe('ChangePassword', () => {
  it('should enable the submit button when passwords are equals and filled and current password is correct', async () => {
    const { getByPlaceholderText, getByTestId } = await renderChangePassword()

    const currentPasswordInput = getByPlaceholderText('Ton mot de passe actuel')
    const passwordInput = getByPlaceholderText('Ton mot de passe')
    const confirmationInput = getByPlaceholderText('Confirmer le mot de passe')

    fireEvent.changeText(currentPasswordInput, 'user@Dfdf56Moi')
    fireEvent.changeText(passwordInput, 'user@AZERTY123')
    fireEvent.changeText(confirmationInput, 'user@AZERTY123')

    // assuming there's only one button in this page
    const continueButton = getByTestId('button-container')

    await waitFor(async () => {
      const background = continueButton.props.style.backgroundColor
      expect(background).toEqual(ColorsEnum.PRIMARY)
    })
  })

  it('should display the matching error when the passwords dont match', async () => {
    const { getByPlaceholderText, getByText } = await renderChangePassword()

    const passwordInput = getByPlaceholderText('Ton mot de passe')
    const confirmationInput = getByPlaceholderText('Confirmer le mot de passe')

    fireEvent.changeText(passwordInput, '123456')
    fireEvent.changeText(confirmationInput, '123456--')

    // assuming there's only one button in this page
    const notMatchingErrorText = getByText('les mots de passe ne concordent pas')

    await waitFor(async () => {
      const color = notMatchingErrorText.props.style[0].color
      expect(color).toEqual(ColorsEnum.ERROR)
    })
  })
  it('should validate PasswordSecurityRules when password is correct', async () => {
    const { getByPlaceholderText, toJSON } = await renderChangePassword()

    const notValidatedRulesSnapshot = toJSON()

    const passwordInput = getByPlaceholderText('Ton mot de passe')
    fireEvent.changeText(passwordInput, 'ABCDefgh1234!!!!')

    await waitForExpect(() => {
      const validatedRulesSnapshot = toJSON()
      expect(notValidatedRulesSnapshot).toMatchDiffSnapshot(validatedRulesSnapshot)
    })
  })
  it.only('', async () => {
    server.use(
      rest.post<ChangePasswordRequest, EmptyResponse>(
        env.API_BASE_URL + '/native/v1/change_password',
        (_req, res, ctx) => res.once(ctx.status(200), ctx.json({}))
      )
    )
    mockedUseSnackBarContext.mockImplementation(() => ({
      displaySuccessSnackBar,
    }))

    const { getByPlaceholderText, getByTestId } = await renderChangePassword()

    const currentPasswordInput = getByPlaceholderText('Ton mot de passe actuel')
    const passwordInput = getByPlaceholderText('Ton mot de passe')
    const confirmationInput = getByPlaceholderText('Confirmer le mot de passe')

    fireEvent.changeText(currentPasswordInput, 'user@Dfdf56Moi')
    fireEvent.changeText(passwordInput, 'user@AZERTY123')
    fireEvent.changeText(confirmationInput, 'user@AZERTY123')

    // assuming there's only one button in this page
    const continueButton = getByTestId('button-container')
    fireEvent.press(continueButton)

    await waitFor(() => {
      expect(displaySuccessSnackBar).toBeCalledWith({
        message: 'Ton mot de passe a été modifié !',
        timeout: SNACK_BAR_TIME_OUT,
      })
    })
  })
})
