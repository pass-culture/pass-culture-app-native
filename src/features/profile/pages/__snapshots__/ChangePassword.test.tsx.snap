// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChangePassword should validate PasswordSecurityRules when password is correct 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -384,11 +384,11 @@
                      \\"flexShrink\\": 1,
                    },
                  ]
                }
                underlineColorAndroid=\\"transparent\\"
-               value=\\"\\"
+               value=\\"ABCDefgh1234!!!!\\"
              />
              <View
                accessible={true}
                focusable={true}
                onClick={[Function onClick]}
@@ -412,11 +412,197 @@
                  height=\\"100%\\"
                  testID=\\"eye-slash\\"
                  width=\\"100%\\"
                >
                  eye-slash-SVG-Mock
+               </View>
+             </View>
+           </View>
+           <View
+             style={
+               Array [
+                 Object {
+                   \\"alignItems\\": \\"flex-start\\",
+                   \\"paddingTop\\": 4,
+                 },
+               ]
+             }
+           >
+             <View
+               style={
+                 Array [
+                   Object {
+                     \\"alignItems\\": \\"center\\",
+                     \\"flexDirection\\": \\"row\\",
+                     \\"maxWidth\\": 500,
+                   },
+                 ]
+               }
+             >
+               <View
+                 height={16}
+                 testID=\\"rule-icon-check\\"
+                 width={16}
+               >
+                 rule-icon-check-SVG-Mock
+               </View>
+               <Text
+                 color=\\"#15884f\\"
+                 style={
+                   Array [
+                     Object {
+                       \\"color\\": \\"#15884f\\",
+                       \\"fontFamily\\": \\"Montserrat-SemiBold\\",
+                       \\"fontSize\\": 12,
+                       \\"lineHeight\\": 16,
+                       \\"paddingLeft\\": 4,
+                     },
+                   ]
+                 }
+               >
+                 12 Caractères
+               </Text>
+             </View>
+             <View
+               style={
+                 Array [
+                   Object {
+                     \\"alignItems\\": \\"center\\",
+                     \\"flexDirection\\": \\"row\\",
+                     \\"maxWidth\\": 500,
+                   },
+                 ]
+               }
+             >
+               <View
+                 height={16}
+                 testID=\\"rule-icon-check\\"
+                 width={16}
+               >
+                 rule-icon-check-SVG-Mock
+               </View>
+               <Text
+                 color=\\"#15884f\\"
+                 style={
+                   Array [
+                     Object {
+                       \\"color\\": \\"#15884f\\",
+                       \\"fontFamily\\": \\"Montserrat-SemiBold\\",
+                       \\"fontSize\\": 12,
+                       \\"lineHeight\\": 16,
+                       \\"paddingLeft\\": 4,
+                     },
+                   ]
+                 }
+               >
+                 1 Majuscule
+               </Text>
+             </View>
+             <View
+               style={
+                 Array [
+                   Object {
+                     \\"alignItems\\": \\"center\\",
+                     \\"flexDirection\\": \\"row\\",
+                     \\"maxWidth\\": 500,
+                   },
+                 ]
+               }
+             >
+               <View
+                 height={16}
+                 testID=\\"rule-icon-check\\"
+                 width={16}
+               >
+                 rule-icon-check-SVG-Mock
+               </View>
+               <Text
+                 color=\\"#15884f\\"
+                 style={
+                   Array [
+                     Object {
+                       \\"color\\": \\"#15884f\\",
+                       \\"fontFamily\\": \\"Montserrat-SemiBold\\",
+                       \\"fontSize\\": 12,
+                       \\"lineHeight\\": 16,
+                       \\"paddingLeft\\": 4,
+                     },
+                   ]
+                 }
+               >
+                 1 Minuscule
+               </Text>
              </View>
+             <View
+               style={
+                 Array [
+                   Object {
+                     \\"alignItems\\": \\"center\\",
+                     \\"flexDirection\\": \\"row\\",
+                     \\"maxWidth\\": 500,
+                   },
+                 ]
+               }
+             >
+               <View
+                 height={16}
+                 testID=\\"rule-icon-check\\"
+                 width={16}
+               >
+                 rule-icon-check-SVG-Mock
+               </View>
+               <Text
+                 color=\\"#15884f\\"
+                 style={
+                   Array [
+                     Object {
+                       \\"color\\": \\"#15884f\\",
+                       \\"fontFamily\\": \\"Montserrat-SemiBold\\",
+                       \\"fontSize\\": 12,
+                       \\"lineHeight\\": 16,
+                       \\"paddingLeft\\": 4,
+                     },
+                   ]
+                 }
+               >
+                 1 Chiffre
+               </Text>
+             </View>
+             <View
+               style={
+                 Array [
+                   Object {
+                     \\"alignItems\\": \\"center\\",
+                     \\"flexDirection\\": \\"row\\",
+                     \\"maxWidth\\": 500,
+                   },
+                 ]
+               }
+             >
+               <View
+                 height={16}
+                 testID=\\"rule-icon-check\\"
+                 width={16}
+               >
+                 rule-icon-check-SVG-Mock
+               </View>
+               <Text
+                 color=\\"#15884f\\"
+                 style={
+                   Array [
+                     Object {
+                       \\"color\\": \\"#15884f\\",
+                       \\"fontFamily\\": \\"Montserrat-SemiBold\\",
+                       \\"fontSize\\": 12,
+                       \\"lineHeight\\": 16,
+                       \\"paddingLeft\\": 4,
+                     },
+                   ]
+                 }
+               >
+                 1 Caractère spécial (!@#$%^&*...)
+               </Text>
              </View>
            </View>
          </View>
          <View
            numberOfSpaces={5}"
`;
